src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -125,12 +125,12 @@ public bool ChildContentRetrieved
         /// <summary>
         /// HTML attributes.
         /// </summary>
-        public TagHelperAttributeList HTMLAttributes { get; }
+        public TagHelperAttributeList HTMLAttributes => _htmlAttributes;
 
         /// <summary>
         /// <see cref="ITagHelper"/> bound attributes and HTML attributes.
         /// </summary>
-        public TagHelperAttributeList AllAttributes { get; }
+        public TagHelperAttributeList AllAttributes => _allAttributes;
```

Comment:
Let's mix this into the https://github.com/aspnet/Razor/pull/683#issuecomment-177061807 offline conversation.